static/module/search.js

```diff
@@ -1,91 +1,81 @@
 import minisearch from 'https://cdn.jsdelivr.net/npm/minisearch@7.1.2/+esm'
 
-export class Search {
-  value = '';
-  data = null;
-
-  constructor(value, data) {
-    this.value = value;
-    this.data = data;
+// Fetches and returns the search data from searchindex.json
+async function fetchSearchData() {
+  const res = await fetch('/static/searchindex.json');
+  if (!res.ok) throw new Error('Failed to fetch search data');
+  return await res.json();
+}
+const data = await fetchSearchData();
+const minisrch = new minisearch({
+  idField: 'name',
+  fields: ['name', 'description', 'author', 'platforms', 'labels'],
+  // keep in sync with searchindex.json.njk
+  storeFields: [
+    'name', 'description', 'author', 'stars', 'platforms', 'labels', 'permalink'
+  ],
+  searchOptions: {
+    boost: { author: 2 },
```

Comment:
Shouldn’t the package name get boosted instead? Or am I missing something?

Reply:
One would think so, but names are a bit special, esp. with fuzzy.  If you start typing "kast", you otherwise get "paste" results first.  Or you type "odat" but get "oat".  Names are not normal words so there is no overlap or fighting with terms usually.

Reply:
Yeah I get that, and I search for my packages as I’m sure you search for yours. But I can also explicitly filter and get exact results. When I search for a language though, many of which have only a few characters in their name, I would prefer if we limit the number of false positive hits in authors or even descriptions, and prefer prioritized hits on the package name.

Reply:
That said, I haven’t tried it yet, so maybe this is indeed better, just asking about the reason for changing the boost.

Reply:
Ok yeah, this works really well in fact. I was worried that with something like `erb` matches on authors would get it in the way, but it's really good actually.

<img width="786" alt="Screenshot 2025-07-01 at 14 57 39" src="https://github.com/user-attachments/assets/078c2456-c092-4f8d-b943-a292cdb3f01c" />